src/core/qgsstringstatisticalsummary.h

```diff
@@ -206,14 +220,16 @@ class CORE_EXPORT QgsStringStatisticalSummary
     Statistics mStatistics;
 
     int mCount;
-    QSet< QString > mValues;
+    QMap< QString, int > mValues;
```

Comment:
Note the datatype change here as an implementation detail. Since strings may be quite large, use a single QMap that stores both the unique keys and the counts. This differs from the numeric implementation in `QgsStatisticalSummary`, where two separate objects are used.

Reply:
Just to be clear, this is an implementation detail, other than little performance change due to `QSet` and `QMap` implementation, there should be no other changes for the users of the API.

Reply:
Would that better be a QHash?

Reply:
Nope, we need them sorted.

Reply:
Since you are there, have a look on this too https://github.com/qgis/QGIS/issues/35433

Reply:
Where exactly?

Reply:
`distinctValues()` is not required, but really nice to return ordered results. The performance gain from `QHash` would be negligible.

Reply:
`distinctVaues()` returns a `QSet` which does not preserve the order.

Reply:
You are right, I got confused with ECMA implementation of `Set` object. I sill believe it would be nice to keep the order for the user, but if you say I would change that.

Reply:
Please do. Values can still be sorted after (even with better control over locale etc. details).

Reply:
I meant keeping the order as they appear in the features. Anyway, it's a `QHash` now.

Reply:
In case of multiple values that are correct answer for `minority` or `majority` statistic, implementation based on `QHash` results in completely random choice.
```
for i in range(10):
  s = QgsStringStatisticalSummary()
  s.calculate(["one", "two", "one", "two"])
  print(s.minority(), s.majority())
```
Using `QMap` is already the solution to this problem in `QgsStatisticalSummary`, so I believe other solutions would bring more noise than benefit. 
Now it is certain, that `minority` and `majority` will return the first occurrence. ("one" is the example above)

Reply:
@m-kuhn it's not always the case that a QHash is a better choice than a QMap for a QString key anyway -- see https://thesmithfam.org/blog/2011/02/12/when-faster-is-actually-slower/. ( I assume that's unchanged for qt 5)